llvm/lib/CodeGen/EarlyIfConversion.cpp

```diff
@@ -561,6 +565,49 @@ bool SSAIfConv::canConvertIf(MachineBasicBlock *MBB, bool Predicate) {
   return true;
 }
 
+/// Apply the target heuristic to decide if the transformation is profitable.
+bool SSAIfConv::isProfitableToConvertIf(
+    const MachineBranchProbabilityInfo *MBPI, TargetSchedModel SchedModel,
+    bool Predicate) {
+  if (!Predicate && TII->shouldConvertPredictableBranches())
```

Comment:
Could you separate out the code movement to a NFC patch, so the functional change of this patch is easier to see?